### PR TITLE
[3.4] Updates to collection versioning.

### PR DESCRIPTION
**Pull request by danielhlarkin**

### CODE REVIEW

**Comment by danielhlarkin** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/1969/

**Comment by danielhlarkin** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/1974/

File: arangod/MMFiles/MMFilesCollection.cpp [hunk 1]
```diff
@@ -3185,6 +3181,12 @@ Result MMFilesCollection::persistLocalDocumentIdsForDatafile(
 }
 
 Result MMFilesCollection::persistLocalDocumentIds() {
+if (_logicalCollection.version() >=
```
Comment (jsteemann) [inline arangod/MMFiles/MMFilesCollection.cpp]: indentation seems wrong here

File: arangod/MMFiles/MMFilesCollection.cpp [hunk 2]
```diff
@@ -3140,6 +3129,13 @@ void MMFilesCollection::removeLocalDocumentId(LocalDocumentId const& documentId,
   }
 }
 
+bool MMFilesCollection::hasAllPersistentLocalIds() const {
+TRI_ASSERT(_hasAllPersistentLocalIds.load() !=
```
Comment (jsteemann) [inline arangod/MMFiles/MMFilesCollection.cpp]: indentation seems wrong here

File: arangod/MMFiles/MMFilesCollection.cpp [hunk 3]
```diff
@@ -3209,10 +3211,28 @@ Result MMFilesCollection::persistLocalDocumentIds() {
...
   TRI_ASSERT(_journals.empty());
+  
+  // mark collection as upgraded so we can avoid re-checking
+  setCurrentVersion();
 
   return Result();
 }
 
+void MMFilesCollection::setCurrentVersion() {
+_logicalCollection.setVersion(
```
Comment (jsteemann) [inline arangod/MMFiles/MMFilesCollection.cpp]: indentation seems wrong here.

**Pull request merged**